shared/trieutil/sparse_merkle.go

```diff
@@ -73,7 +100,22 @@ func BranchIndices(merkleIndex int, depth int) []int {
 	return indices
 }
 
-// Root of the Merkle trie.
+// HashTreeRoot of the Merkle trie as defined in the deposit contract.
+//  Spec Definition:
+//   sha256(concat(node, self.to_little_endian_64(self.deposit_count), slice(zero_bytes32, start=0, len=24)))
```

Comment:
i don't believe we need this - we should simply compute hash tree root with capacity of the list of deposit data instead